solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -114,4 +114,65 @@ print(ans)
 ```
 
 </PySection>
+
+	
+<JavaSection>
+	
+```java
+import java.util.*;
+import java.io.*;
+
+public class GCDBoard {
+    static final int maxN = 100000 + 5;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        int N = Integer.parseInt(br.readLine());
+        StringTokenizer st = new StringTokenizer(br.readLine());
+
+        int[] arr = new int[maxN];
+        for (int x = 1; x <= N; x++) {
```

Comment:
is 1-indexing necessary?

Reply:
Yes. In the answer I do prefix[x-1] so I need 0 to be a buffer. BTW, the internal sol for C++ also uses one indexing